File: app/src/main/java/iammert/com/androidarchitecture/ui/detail/MovieDetailViewModel.java
package iammert.com.androidarchitecture.ui.detail;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import javax.inject.Inject;

import iammert.com.androidarchitecture.data.MovieRepository;
import iammert.com.androidarchitecture.data.Resource;
import iammert.com.androidarchitecture.data.local.entity.MovieEntity;

/**
 * Created by mertsimsek on 21/05/2017.
 */

public class MovieDetailViewModel extends ViewModel{
    private final LiveData<Resource<MovieEntity>>  movieDetail = new MutableLiveData<>();
    private final MovieRepository movieRepository;

    @Inject
    public MovieDetailViewModel(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    public LiveData<MovieEntity> getMovie(int id){
        return movieRepository.getMovie(id);
    }
}

File: app/src/main/java/iammert/com/androidarchitecture/ui/detail/MovieDetailViewModel.kt
package iammert.com.androidarchitecture.ui.detail

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import iammert.com.androidarchitecture.data.MovieRepository
import iammert.com.androidarchitecture.data.local.entity.MovieEntity
import javax.inject.Inject

class MovieDetailViewModel @Inject
constructor(val movieRepository: MovieRepository) : ViewModel() {

    fun getMovie(id: Int): LiveData<MovieEntity> =
            movieRepository.getMovie(id)
}
